### PR TITLE
Update lint rules for version 1.2.0

### PR DESCRIPTION
**Pull request by raichibald**

### CODE REVIEW

**Review by maxpelna — COMMENTED** [review]
can we also add [avoid_function_literals_in_foreach_calls] https://dart.dev/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,53 @@
+# 1.2.0
+- **refactor:** remove `dart_code_metrics` dependency
+- **refactor:** remove [banned-usage](https://dcm.dev/docs/rules/common/banned-usage/) `RichText` identifier in favour of [prefer-text-rich](https://dcm.dev/docs/rules/flutter/prefer-text-rich/)
+- **refactor:** increase `max-identifier-length` of [prefer-correct-identifier-length](https://dcm.dev/docs/rules/common/prefer-correct-identifier-length/) to 40
+
+- **fix:** set `ignore-blocs` to `true` for `dispose-fields` (BlocProvider automatically handles disposal of the Bloc)
```
Comment (darjaorlova) [inline CHANGELOG.md:6]: Important note: only those blocs, that were created via `create` method from `BlocProvider`, not via `BlocProvider.value`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+          - path: 'data/.*_request_body.dart'
+            pattern: 'RequestBody$'
+          - path: 'data/.*_response.dart'
+            pattern: 'Response$'
+       ```
+    - [avoid-unnecessary-reassignment](https://dcm.dev/docs/rules/common/avoid-unnecessary-reassignment/)
+    - [prefer-correct-error-name](https://dcm.dev/docs/rules/common/prefer-correct-error-name/)
+      ```yaml
+        allowed-names:
+          - ex
```
[the remark this answers is not in the document]
Comment (raichibald) [reply CHANGELOG.md:34]: You are silly 😝 😅

**Comment by raichibald** [comment]
> can we also add [avoid_function_literals_in_foreach_calls] https://dart.dev/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal

Added new Dart lint rules:
    - avoid_function_literals_in_foreach_calls
    - prefer_expression_function_bodies
    - eol_at_end_of_file
    - always_use_package_imports

**Pull request merged**